### ISSUE TITLE
`reportInvalidTypeForm` false positive on type alias to `Never`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#490]
```py
from typing import Never

Foo = Never
value: Foo # error: reportInvalidTypeForm
```
introduced upstream in 1.1.371

**Comment by DetachHead** [comment]
seems to be partially fixed in 1.1.379, but it's still broken inside functions:

```py
from typing import Never

Foo = Never

value: Foo # no error anymore

def foo():
    value: Foo # error: reportInvalidTypeForm
```
